test/integration/adapters.js

```diff
@@ -27,23 +26,258 @@ function _attachListeners (done, runner) {
+  })
+}
+
+function _testTestOnEnd (index, statusMsg, errorMsg) {
+  it('should have been emitted on testEnd event', function () {
+    expect(collectedData[index][0]).to.be.equal('testEnd')
+  })
+
+  it('should have the same name as on start', function () {
+    expect(collectedData[index][1].testName).to.be
+      .equal(refData[index][1].testName)
+  })
+
+  it('should contain the name of the suite parent', function () {
+    expect(collectedData[index][1].suiteName).to.be
+      .equal(refData[index][1].suiteName)
+  })
+
+  it('should contain status as ' + statusMsg, function () {
+    expect(collectedData[index][1].status).to.be
+      .equal(refData[index][1].status)
+  })
+
+  if (statusMsg === 'skipped') {
+    it('should have no runtime', function () {
+      expect(collectedData[index][1].runtime).to.be
+        .equal(refData[index][1].runtime)
+    })
+  } else {
+    it('should contain its runtime', function () {
+      expect(collectedData[index][1].runtime).to.be
+        .closeTo(refData[index][1].runtime, 1)
+    })
+  }
+
+  it('should have ' + errorMsg, function () {
+    expect(collectedData[index][1].errors).to.be.deep
+      .equal(refData[index][1].errors)
+  })
+}
+
+function _testSimpleSuite (index) {
+  it('should have no other child suites', function () {
+    expect(collectedData[index][1].childSuites).to.have
+      .lengthOf(refData[index][1].childSuites.length)
+  })
+
+  it('should contain only one test', function () {
+    expect(collectedData[index][1].tests).to.have
+      .lengthOf(refData[index][1].tests.length)
+  })
+}
+
+function _testComplexSuite (index) {
+  it('should have no other child suites', function () {
+    expect(collectedData[index][1].childSuites).to.have
+      .lengthOf(refData[index][1].childSuites.length)
+  })
+
+  it('should contain only multiple tests', function () {
+    expect(collectedData[index][1].tests).to.have
+      .lengthOf(refData[index][1].tests.length)
+  })
+}
+
+function _testNestedSuite (index) {
+  it('should have one child suite', function () {
+    expect(collectedData[index][1].childSuites).to.have
+      .lengthOf(refData[index][1].childSuites.length)
+  })
+
+  it('should contain one test', function () {
+    expect(collectedData[index][1].tests).to.have
+      .lengthOf(refData[index][1].tests.length)
+  })
+}
+
+/**
+ * @param index {Integer}
+ * @param type {Char} - Describing what type of suite is under testing.
+ *
+ * s - simple suite with only one test and no child suites.
+ * c - complex suite with multiple tests and no child suites.
+ * n - nested suite with one test and one child suite.
+ */
+function _testSuite (index, type) {
+  it('should be emitted on suiteStart event', function () {
+    expect(collectedData[index][0]).to.be.equal('suiteStart')
+  })
+
+  it('should have a name', function () {
+    expect(collectedData[index][1].suiteName).to.be
+      .equal(refData[index][1].suiteName)
+  })
+
+  switch (type) {
```

Comment:
This is not a good idea. At least use something readable ("simple" instead of "s"), better yet, pass functions around. So instead of `_testSuite(3, 's')`, do something like `_testSuite(3, _testSimpleSuite)`, then call that function with the `index` argument.